### PR TITLE
Added standalone diagram-builder haskell package.

### PR DESCRIPTION
**Pull request by jraygauthier**
Simple wrapper over `haskellng`'s `diagram-builder`
package that allows for simpler install.

### CODE REVIEW

File: pkgs/tools/graphics/diagrams-builder/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  }
+  ­~~~
+*/
+
+{ stdenv, ghcWithPackages, makeWrapper, diagrams-builder, extraPackages ? (self: []) }:
+let
+  # Used same technique as for the yiCustom package.
+  w = ghcWithPackages 
+    (self: [ diagrams-builder ] ++ extraPackages self);
+  wrappedGhc = w.override { ignoreCollisions = true; };
```
Comment (peti) [inline pkgs/tools/graphics/diagrams-builder/default.nix]: Why is it necessary to ignore file collisions?

File: pkgs/tools/graphics/diagrams-builder/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  ghcVersion = w.version;
+in
+stdenv.mkDerivation {
+  name = "diagrams-builder";
+  buildInputs = [ makeWrapper ];
+  buildCommand = ''
```
Comment (peti) [inline pkgs/tools/graphics/diagrams-builder/default.nix]: Commit c449a93 adds a `postBuild` hook to the normal `ghcWithPackages` function that can be used to wrap programs, etc. Maybe the implementation of this function can become even simpler by using that mechanism?

File: pkgs/tools/graphics/diagrams-builder/default.nix (hunk 1 above)
Comment (jraygauthier) [inline pkgs/tools/graphics/diagrams-builder/default.nix]: Most likely an artifact of times past. Will have to test this.

File: pkgs/tools/graphics/diagrams-builder/default.nix (hunk 2 above)
Comment (jraygauthier) [inline pkgs/tools/graphics/diagrams-builder/default.nix]: I'm not sure I can see the potential yet. Do you have an example?

I can definitely make the whole thing more concise by moving 

~~~
makeWrapper \
"${diagrams-builder}/bin/diagrams-builder-svg" "$out/bin/diagrams-builder-svg" \
 --set NIX_GHC ${wrappedGhc}/bin/ghc \
 --set NIX_GHC_LIBDIR ${wrappedGhc}/lib/ghc-${ghcVersion}
~~~

into a function parametrized with the executable suffix that differs.

**Comment by jraygauthier** [comment]
As discussed.

**Pull request merged**